msg-gateway/src/main/java/io/fabric8/msg/gateway/brokers/impl/KubernetesBrokerControl.java

```diff
@@ -112,44 +99,26 @@ public void setArtemisName(String artemisName) {
         this.artemisName = artemisName;
     }
 
-    public String getPortName() {
-        return portName;
+
+    public String getArtemisPort() {
+        return artemisPort;
     }
 
-    public void setPortName(String portName) {
-        this.portName = portName;
+    public void setArtemisPort(String artemisPort) {
+        this.artemisPort = artemisPort;
     }
 
+
     protected void lookupBrokers() {
-        System.err.println("LOOKUP BROKERS CALLED");
         try {
-
-            Endpoints endpoints = kubernetesClient.endpoints().inNamespace(getNamespace()).withName(getArtemisName()).get();
-    System.err.println("ENDPOINTS = " + endpoints);
-
             HashSet<ArtemisClient> set = new HashSet<>();
-            if (endpoints != null) {
-                for (EndpointSubset subset : endpoints.getSubsets()) {
-                    if (subset.getPorts().size() == 1) {
-                        EndpointPort port = subset.getPorts().get(0);
-                        for (EndpointAddress address : subset.getAddresses()) {
-                            ArtemisClient artemisClient = new ArtemisClient(address.getIp(), port.getPort());
-                            set.add(artemisClient);
-                        }
-                    } else {
-                        for (EndpointPort port : subset.getPorts()) {
-                            if (Utils.isNullOrEmpty(portName) || portName.endsWith(port.getName())) {
-                                for (EndpointAddress address : subset.getAddresses()) {
-                                    ArtemisClient artemisClient = new ArtemisClient(address.getIp(), port.getPort());
-                                    set.add(artemisClient);
-                                }
-                            }
-                        }
-                    }
-                }
-            }
+            Set<String> brokers = KubernetesHelper.lookupServiceInDns(getArtemisName());
+            brokers.forEach(broker -> {
+                    System.err.println("Broker = " + broker);
```

Comment:
This is just a stub - @jstrachan is going to implement it properly - with UI support for defining the cluster etc